lib/Workflow/Manager.php

```diff
@@ -333,4 +334,74 @@ public function getTransitionByName(string $workflowName, string $transitionName
 
         return null;
     }
+
+    /**
+     * Compatibility Fix: forces an initial place being set (and stored) if the current place is empty.
+     * We cannot apply a regular transition b/c it would be considered invalid by the state machine.
+     *
+     * As of Symfony 4.4.8 built-in implementations of @see \Symfony\Component\Workflow\MarkingStore\MarkingStoreInterface
+     * use strict `null` comparison when retrieving the current marking. This will throw exceptions for elements
+     * stored with empty marking despite an initial state was configured in prior versions of Pimcore.
+     *
+     * @param Asset|Concrete|PageSnippet $subject
+     *
+     * @return bool true if initial state was applied
+     *
+     * @throws \Exception
+     */
+    public function ensureBackwardCompatibleInitialPlace(string $workflowName, $subject): bool
+    {
+        if (!$workflow = $this->getWorkflowIfExists($subject, $workflowName)) {
+            return false;
+        }
+
+        if (!$markingStore = $workflow->getMarkingStore()) {
+            return false;
+        }
+
+        // check that the subject has a non-empty place
+        $initialPlaces = $this->getInitialPlacesForWorkflow($workflow);
+        $markingObject = $markingStore->getMarking($subject);
+        foreach ($markingObject->getPlaces() as $placeName => $nbToken) {
+            if ('' !== $placeName) {
+                continue;
+            }
+
+            // fill empty place with initial place, if any
+            $markingObject->unmark($placeName);
+            foreach ($initialPlaces as $initialPlace) {
+                $markingObject->mark($initialPlace);
+            }
+
+            $markingStore->setMarking($subject, $markingObject);
+
+            // StateTableMarkingStore handles persistence of it's own
+            if ($markingStore instanceof StateTableMarkingStore === false) {
+                $wasOmitMandatoryCheck = $subject->getOmitMandatoryCheck();
+                $subject->setOmitMandatoryCheck(true);
+                $subject->save();
+                $subject->setOmitMandatoryCheck($wasOmitMandatoryCheck);
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getInitialPlacesForWorkflow(Workflow $workflow): array
+    {
+        $initialPlaces = [];
+        $definition = $workflow->getDefinition();
+
+        if(method_exists($definition, 'getInitialPlaces')) {
```

Comment:
I would prefer to use `$definition->getInitialPlace()` for Pimcore 6 and switch to `$definition->getInitialPlaces()` with Pimcore 7 - when we need to get rid of all deprecation messages for symfony 5 compatibility anyway. 

This construct here blows up the code and we will never find it again when we are starting with Pimcore 7.

Reply:
I get your point, but does it fit into the concept of [multiple initial places](https://symfony.com/blog/new-in-symfony-4-3-workflow-improvements#allow-to-configure-many-initial-places)?

`$definition->getInitialPlace()` will return only the 1st element. 🤔

```
    /**
     * @deprecated since Symfony 4.3. Use getInitialPlaces() instead.
     */
    public function getInitialPlace(): ?string
    {
        /// ...
        return reset($this->initialPlaces);
    }
```
(on the other hand I'm not quite sure if Pimcore 6 supports multiple initial places)

Reply:
good point ... multiple initial places should be supported since #5541

then I guess we need to do it the way it is 👍